reconcile/openshift_tekton_resources.py

```diff
@@ -0,0 +1,353 @@
+import sys
+import logging
+import json
+from typing import Any, Iterable, Optional, Union
+
+import yaml
+import jinja2
+
+from reconcile import queries
+from reconcile import openshift_base as ob
+from reconcile.status import ExitCodes
+from reconcile.utils import gql
+from reconcile.utils.defer import defer
+from reconcile.utils.openshift_resource import OpenshiftResource as OR
+from reconcile.utils.semver_helper import make_semver
+
+LOG = logging.getLogger(__name__)
+QONTRACT_INTEGRATION = 'openshift-tekton-resources'
+QONTRACT_INTEGRATION_VERSION = make_semver(0, 1, 0)
+
+# it must be a single character due to resource max length
+OBJECTS_PREFIX = 'o'
+RESOURCE_MAX_LENGTH = 63
+
+# Defaults
+DEFAULT_DEPLOY_RESOURCES = {'requests': {'cpu': '50m',
+                                         'memory': '200Mi'},
+                            'limits': {'cpu': '200m',
+                                       'memory': '300Mi'}}
+SAAS_FILES_QUERY = """
+{
+  saas_files: saas_files_v2 {
+    path
+    name
+    configurableResources
+    pipelinesProvider {
+      name
+      provider
+    }
+    deployResources {
+      requests {
+        cpu
+        memory
+      }
+      limits {
+        cpu
+        memory
+      }
+    }
+  }
+}
+"""
+
+
+class OpenshiftTektonResourcesNameTooLongError(Exception):
+    pass
+
+
+class OpenshiftTektonResourcesBadConfigError(Exception):
+    pass
+
+
+# Builds a list of v2 saas files from qontract-server
+def get_saas_files(saas_file_name: Optional[str]) -> list[dict[str, Any]]:
+    saas_files = [
+        s for s in gql.get_api().query(SAAS_FILES_QUERY)['saas_files']
+        if s.get('configurableResources')]
+
+    if saas_file_name:
+        saas_file = None
+        for s in saas_files:
+            if s['name'] == saas_file_name:
+                saas_file = s
+                break
+
+        return [saas_file] if saas_file else []
+
+    return saas_files
+
+
+def fetch_tkn_providers(saas_files: Iterable[dict[str, Any]]) \
```

Comment:
If I were super pedantic this'd be `Iterable[Mapping[str, Any]]`.

Reply:
I've consciously avoided using mappings since these functions are internal to this module. I can see the benefit of using mappings in a library to be used externally, though.